In [5]:
import pandas as pd

df = pd.read_excel("Sample - Superstore.xls")

# **Analyse du dataframe**

Affiche les 3 premières lignes du dataframe (matrice de taille (3, 21)) avec la fonction "head()"

In [7]:
h = df.head(3)

print(h)

   Row ID        Order ID Order Date  Ship Date     Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11  Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11  Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16  Second Class    DV-13045   

     Customer Name    Segment        Country         City  ... Postal Code  \
0      Claire Gute   Consumer  United States    Henderson  ...       42420   
1      Claire Gute   Consumer  United States    Henderson  ...       42420   
2  Darrin Van Huff  Corporate  United States  Los Angeles  ...       90036   

   Region       Product ID         Category Sub-Category  \
0   South  FUR-BO-10001798        Furniture    Bookcases   
1   South  FUR-CH-10000454        Furniture       Chairs   
2    West  OFF-LA-10000240  Office Supplies       Labels   

                                        Product Name   Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.96   

Affiche les 3 dernières lignes du dataframe (matrice de taille (3, 21)) avec la fonction "tail()"

In [9]:
t = df.tail(3)

print(t)

      Row ID        Order ID Order Date  Ship Date       Ship Mode  \
9991    9992  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9992    9993  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9993    9994  CA-2017-119914 2017-05-04 2017-05-09    Second Class   

     Customer ID Customer Name   Segment        Country         City  ...  \
9991    DB-13060   Dave Brooks  Consumer  United States   Costa Mesa  ...   
9992    DB-13060   Dave Brooks  Consumer  United States   Costa Mesa  ...   
9993    CC-12220  Chris Cortes  Consumer  United States  Westminster  ...   

     Postal Code  Region       Product ID         Category Sub-Category  \
9991       92627    West  TEC-PH-10003645       Technology       Phones   
9992       92627    West  OFF-PA-10004041  Office Supplies        Paper   
9993       92683    West  OFF-AP-10002684  Office Supplies   Appliances   

                                           Product Name    Sales  Quantity  \
9991                           

Afiiche les dimensions du dataframe (matrice de taille (9994, 21)) avec la fonction "shape"

In [11]:
s = df.shape

print(s)

(9994, 21)


Affiche la liste des index du dataframe avec la fonction columns.
L'option "to_list()" permet de renvoyer un array (une liste en python).

In [14]:
c = df.columns

print(c)
print(c.to_list())

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')
['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode', 'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State', 'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category', 'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit']


Affiche les différents types pour les index avec la fonction "dtypes"

In [16]:
d = df.dtypes

print(d)

Row ID                    int64
Order ID                 object
Order Date       datetime64[ns]
Ship Date        datetime64[ns]
Ship Mode                object
Customer ID              object
Customer Name            object
Segment                  object
Country                  object
City                     object
State                    object
Postal Code               int64
Region                   object
Product ID               object
Category                 object
Sub-Category             object
Product Name             object
Sales                   float64
Quantity                  int64
Discount                float64
Profit                  float64
dtype: object


Affiche